static/app/views/settings/project/server-side-sampling/modals/specificConditionsModal/index.tsx

```diff
@@ -153,12 +154,18 @@ export function SpecificConditionsModal({
       ? rules.map(existingRule => (existingRule.id === rule.id ? newRule : existingRule))
       : [...rules, newRule];
 
+    // Make sure that a uniform rule is always send in the last position of the rules array
+    const [uniformRule, specificRules] = partition(newRules, isUniformRule);
```

Comment:
I'm not sure if partitioning conditions into "uniform" and "specific" will fix all the cases. Imagine rules

* `release=foo -> 20%`
* `release=foo environment=bar -> 100%`
* `all -> 1%`

`all -> 1%` would be moved to the bottom by this logic, but `release=foo -> 20%` would still precede the more specific `release=foo environment=bar -> 100%`.

Automatically determining which rule should have precedence is hard (which is why we don't even attempt to do so in Relay). So to be honest I would just give the user a rearrangeable list of rules and make it clear to them that the first matching rule wins.

Reply:
Yes, but we are doing it. In the UI, the uniform rules always need to be fixed at the bottom and cannot be dragged (else operator), while the other rules can be dragged and the UI shows for each one an operator "if and else if". The idea behind the operator was to display the order of priority

Reply:
**if** release=foo -> 20%
**else if** release=foo environment=bar -> 100%
**else** all -> 1

Reply:
OK, so it's a special case reordering for `else`. That makes sense I guess.